nilearn/surface/surface.py

```diff
@@ -0,0 +1,641 @@
+    sample_locations = _sample_locations(
+        mesh, affine, kind=kind, radius=radius, n_points=n_points)
+    n_vertices, n_points, img_dim = sample_locations.shape
+    grid = [np.arange(size) for size in images[0].shape]
+    interp_locations = np.vstack(sample_locations)
+    masked = _masked_indices(interp_locations, images[0].shape, mask=mask)
+    # loop over images rather than building a big array to use less memory
+    all_samples = []
+    for img in images:
+        interpolator = interpolate.RegularGridInterpolator(
+            grid, img,
+            bounds_error=False, method='linear', fill_value=None)
+        samples = interpolator(interp_locations)
+        # if all samples around a mesh vertex are outside the image,
+        # there is no reasonable value to assign to this vertex.
+        # in this case we return NaN for this vertex.
+        samples[masked] = np.nan
+        all_samples.append(samples)
+    all_samples = np.asarray(all_samples)
+    all_samples = all_samples.reshape((len(images), n_vertices, n_points))
+    texture = np.nanmean(all_samples, axis=2)
+    return texture
+
+
+def vol_to_surf(img, surf_mesh,
+                radius=3., kind='line', interpolation='nearest',
+                n_samples=None, mask_img=None):
+    """Extract surface data from a Nifti image.
+
+    Parameters
+    ----------
+
+    img : Niimg-like object, 3d or 4d.
+        See http://nilearn.github.io/manipulating_images/input_output.html
+
+    surf_mesh : str or numpy.ndarray
+        Either a file containing surface mesh geometry (valid formats
+        are .gii or Freesurfer specific files such as .orig, .pial,
+        .sphere, .white, .inflated) or a list of two Numpy arrays,
+        the first containing the x-y-z coordinates of the mesh
+        vertices, the second containing the indices (into coords)
+        of the mesh faces.
+
+    radius : float, optional (default=3.).
+        The size (in mm) of the neighbourhood from which samples are drawn
+        around each node.
+
+    kind : {'line', 'ball'}
+        The strategy used to sample image intensities around each vertex.
+
+        - 'line' (the default):
+            samples are regularly spaced along the normal to the mesh, over the
+            interval [- `radius`, + `radius`].
+            (sometimes called thickness sampling)
+        - 'ball':
+            samples are regularly spaced inside a ball centered at the mesh
+            vertex.
+
+    interpolation : {'linear', 'nearest'}
+        How the image intensity is measured at a sample point.
+
+        - 'nearest' (the default):
+            Use the intensity of the nearest voxel.
+        - 'linear':
+            Use a trilinear interpolation of neighboring voxels.
+
+    n_samples : int or None, optional (default=None)
+        How many samples are drawn around each vertex and averaged. If
+        ``None``, use a reasonable default for the chosen sampling strategy
+        ('ball' or 'line').
```

Comment:
While we are at it. I think it is a better option to also state default samples for 'line'.